floc.bs

```diff
@@ -1 +1,178 @@
+
+  The user agent should offer a dedicated permission setting for the user to disallow sites from being included for interest cohort calculations.
+
+  <h4 id="access-eligibility">Permission to access the interest cohort</h4>
+  The page can restrict itself or subframes from accessing the interest cohort through the "<code><a href="#interest-cohort-policy-controlled-feature">interest-cohort</a></code>" [=policy-controlled feature=].  [[!PERMISSIONS-POLICY]]
+
+  <a href="#the-api">The API</a> will return a rejected promise if the user has specifically disallowed the site from accessing the [=interest cohort=].
+
+  <h4 id="private-browsing-mode">Private browsing / Incognito mode</h4>
+  The interest cohort computation <a href="#cohort-assignment-algorithm">algorithm</a> and the {{Document/interestCohort()}} API methods are applicable to the private browsing mode as well. That is, if the private browsing mode doesn't save history at all, the "information from the browsing history" is expected to just be an empty set.
+
+  <h4 id="adoption-phase">Adoption phase</h4>
+  To make the adoption easier, the user agent may relax the opt-in requirement while third-party cookies still exist. For example, pages with ads resources are an approximation of the pages that are going to opt-in to interest cohort computation in the long run. Thus, at the adoption phase, the page can be eligible to be included in the interest cohort computation if there are ads resources in the page, OR if <a href="#the-api">the API</a> is used.
+
+  Additionally, during the adoption phase, the browser can use the existing cookie settings to approximate the interest cohort permission setting. For example, a page is not allowed to contribute to the interest cohort calculation if cookies are disallowed for that site; when cookies are cleared, previous page visits should not be used for interest cohort computation; accessing to the interest cohort within a {{Document}} should be denied if cookie access is not allowed in the document, or when third-party cookies are disallowed in general.
+
+  <h3 id="sensitive-information">Sensitive information</h3>
+  An [=interest cohort=] might reveal sensitive information. As a first mitigation, the browser should remove sensitive categories from its data collection. But this does not mean sensitive information can’t be leaked. Some people are sensitive to categories that others are not, and there is no globally accepted notion of sensitive categories.
+
+  Cohorts could be evaluated for fairness by measuring and limiting their deviation from population-level demographics with respect to the prevalence of sensitive categories, to prevent their use as proxies for a sensitive category. However, this evaluation would require knowing how many individual people in each cohort were in the sensitive categories, information which could be difficult or intrusive to obtain.
```

Comment:
As an approximation, the browser could use a mechanism for recognizing which web pages are in sensitive categories.

Reply:
Done.